service/history/workflowResetor.go

```diff
@@ -501,13 +511,18 @@ func (w *workflowResetorImpl) replayReceivedSignals(ctx context.Context, receive
 		continueRunID = ""
 
 		var nextPageToken []byte
+		var shardID *int
+		if w.eng != nil && w.eng.shard != nil {
```

Comment:
same as above. Our initialization has made sure it won't happen. Otherwise it should just panic without making any process.

Reply:
removed